src/Module.h

```diff
@@ -187,25 +190,24 @@ class Module {
 /** Link a set of modules together into one module. */
 Module link_modules(const std::string &name, const std::vector<Module> &modules);
 
-/** Create an object file containing the Halide runtime for a given target. For
- * use with Target::NoRuntime. Standalone runtimes are only compatible with
- * pipelines compiled by the same build of Halide used to call this function. */
-void compile_standalone_runtime(const std::string &object_filename, Target t);
+using TelemetryFactory = std::function<std::unique_ptr<Internal::Telemetry>(const std::string &fn_name, const Target &target)>;
 
 /** Create an object and/or static library file containing the Halide runtime
  * for a given target. For use with Target::NoRuntime. Standalone runtimes are
  * only compatible with pipelines compiled by the same build of Halide used to
  * call this function. Return a map with just the actual outputs filled in
  * (typically, Output::object and/or Output::static_library).
  */
-std::map<Output, std::string> compile_standalone_runtime(const std::map<Output, std::string> &output_files, Target t);
+std::map<Output, std::string> compile_standalone_runtime(const std::map<Output, std::string> &output_files, Target t,
```

Comment:
I'm confused by this signature change. I don't think it's legal to compile a standalone runtime to anything other than a single object file, is it? Also telemetry doesn't seem like it makes any sense for a standalone runtime. If either overload was deleted I would think it would be this one. We should at least preserve the existing one - IMO it's the one with the signature that make sense for users.

Reply:
It's ~always been legal to compile a runtime into static_library form too.

That said, removing this overload is probably more sensible to do separately; it was done here to test the telemetry for runtimes, which (as you mentioned) doesn't make a lot of sense. I'll drop those changes.

Reply:
Oh right, static library too.

Reply:
Removed the runtime changes and Telemetry, also rebased to master